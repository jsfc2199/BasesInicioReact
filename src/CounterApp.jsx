import PropTypes from "prop-types";
import React, { useState } from "react";

export const CounterApp = ({ value }) => {

  //Primer hook useState
  const [counter, setCounter] = useState(value) //sirve para modificar el estado de algo, tiene un valor inicial

  //funciones
  function sumarUno(event) {
    setCounter(counter + 1)

    // setCounter((c) => c+1) otra forma de hacerlo
  }

  function restarUno() {
    setCounter(counter - 1)
  }

  function reset() {
    setCounter(value)
  }

//   lo mismo pero como arrow function
//   const sumarUno = (event) => {
//     console.log(event);
//   }

  return (
    <>
      <h1>Counter App</h1>

      {/* uso el counter del useState */}
      <h2>{counter}</h2>

      {/* evento onClick */}
      <button onClick={(event) => sumarUno(event)}>+1</button>
      <button onClick={(event) => restarUno(event)}>-1</button>
      <button onClick={reset}>reset</button>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};
