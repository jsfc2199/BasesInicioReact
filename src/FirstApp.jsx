import React from "react";

//si hay objetos que no se van a cambiar para nada ni hacen parte de hooks podemos sacarlo del functional component
const objeto = {
  title: "esto es un título",
  objeto: "esto es un objeto",
};

const getResult = () => {
  return 4 + 4;
};
export const FirstApp = () => {
  //variables
  const nombre = "Juan";

  return (
    // si mandamos <> </> es un fragment
    //también se puede usar <Fragment></Fragment>
    <>
      <div>FirstApp</div>
      <p>párrafo</p>

      {/* imprimir variables en el html */}
      <h3>Tu nombre es: {nombre}</h3>
      <h3>La suma de 1+1 es: {1 + 1}</h3>
      <div>{objeto.title}</div>
      <div>{objeto.objeto}</div>
      <div>el resultado de getResult es: {getResult()}</div>

      {/* Booleanos no se imprimen */}
      <div>el resultado de getResult es: {false}</div>

    </>
  );
};
