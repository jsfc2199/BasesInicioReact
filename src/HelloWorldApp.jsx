import React from 'react'

const HelloWorldApp = () => {
  return (
    <h1>
      Hello Word
    </h1>
  )
}

export default HelloWorldApp
