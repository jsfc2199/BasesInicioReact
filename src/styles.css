html, body {
    background-color: rgb(0, 0, 0);
    color: white;
    font-size: 1.3rem;
}