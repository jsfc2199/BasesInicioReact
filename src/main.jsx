/* eslint-disable no-unused-vars */
//jsx es html + js
//la app de react siempre cuenta con estos dos imports
import React from "react";
import ReactDOM from "react-dom/client";
import HelloWorldApp from './HelloWorldApp';
import { FirstApp } from './FirstApp';
import {CounterApp} from './CounterApp';

import './styles.css' //se debe importar para tener estilos globales

ReactDOM.createRoot(document.getElementById("root")).render(
  //miramos que el index.html tiene un id root, all√≠ se renderiza todo y renderizamos a App
  <React.StrictMode>
    {/* <HelloWorldApp/> */}

    {/* enviamos props */}
    {/* <FirstApp title="Esto es una prop" subtitle={123}/> */}
    <CounterApp value={123}/>
  </React.StrictMode>
);
