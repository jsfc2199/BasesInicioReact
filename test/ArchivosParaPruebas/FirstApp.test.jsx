import { render } from "@testing-library/react";
import { FirstApp } from "../../src/FirstApp";

describe("Pruebas en first app", () => {
  test("Debe hacer match con el snapshot", () => {
    const title = "Hola";
    const subtitle = 123;

    const { container } = render(
      <FirstApp title={title} subtitle={subtitle} />
    );

    expect(container).toMatchSnapshot(); //foto de lo que hay
  });

  
  test("Debe de mostrar el titulo en un h2", () => {
    const title = "Hola";
    const subtitle = 123;

    const { container, getByText } = render(
      <FirstApp title={title} subtitle={subtitle} />
    );

    expect(getByText(title)).toBeTruthy()

    //no recomendado
    const h1 = container.querySelector('h2')
    expect(h1.innerHTML).toBe(title)

  });
});
