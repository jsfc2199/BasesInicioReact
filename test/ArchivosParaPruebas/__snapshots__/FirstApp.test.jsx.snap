// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en first app Debe hacer match con el snapshot 1`] = `
<div>
  <h2>
    Hola
  </h2>
  <h2>
    123
  </h2>
  <div>
    FirstApp
  </div>
  <p>
    párrafo
  </p>
  <h3>
    Tu nombre es: 
    Juan
  </h3>
  <h3>
    La suma de 1+1 es: 
    2
  </h3>
  <div>
    esto es un título
  </div>
  <div>
    esto es un objeto
  </div>
  <div>
    el resultado de getResult es: 
    8
  </div>
  <div>
    el resultado de un booleano es: 
  </div>
</div>
`;
